src/runtime/virtcontainers/clh.go

```diff
@@ -1401,6 +1421,11 @@ func (clh *cloudHypervisor) isClhRunning(timeout uint) (bool, error) {
 			clh.Logger().WithError(err).Warning("clh.VmmPingGet API call failed")
 		}
 
+		err = syscall.Kill(pid, syscall.Signal(0))
+		if err != nil {
+			return false, nil
+		}
+
```

Comment:
IIUC this is the `Prevent a 10 second timeout` optimization. What about doing this at the beginning of the block ? This will allow to drop the same check that is done before the loop.

Also, this would deserve its own commit.

Reply:
Correct, i'll fix it.

Reply:
Fixed.